src/Symfony/Component/Workflow/Debug/ListenerExtractor.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow\Debug;
+
+use Symfony\Component\ErrorHandler\ErrorRenderer\FileLinkFormatter;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\Workflow\Definition;
+use Symfony\Component\Workflow\EventListener\GuardExpression;
+use Symfony\Component\Workflow\EventListener\GuardListener;
+use Symfony\Component\Workflow\Transition;
+
+/**
+ * @internal
```

Comment:
I using the WorkflowDataCollector requires instantiating that class, it cannot really be internal, as its instantiation is necessary in code consuming the WorkflowDataCollector.

My suggestion though would be to keep it internal, but making it an internal implementation detail of the WorkflowDataCollector (i.e. keeping the existing constructor signature taking the event dispatcher and the file link formatter)